### ISSUE TITLE
bug for softmaxwithloss operator

### ISSUE THREAD

**Issue by ygwangmsra** [pytorch/pytorch#11885]
I found that the ignore label does not work for the operator: softmaxwithloss in [https://github.com/pytorch/pytorch/blob/master/caffe2/operators/softmax_with_loss_op.cc](url), which means that line 151 #define DONT_CARE (-1) does not make sense

On the other hand, the operator: spatialsoftmaxwithloss in [https://github.com/pytorch/pytorch/blob/master/caffe2/operators/spatial_softmax_with_loss_op.cc](url) does not have the bug, we can igore the label -1

Hope to fix the bug.
